### DISCUSSION TITLE
Where doest nextjs store cache.

### DISCUSSION THREAD

**Discussion by crazyxhz** [vercel/next.js#47632] [Help]
### Summary

nextjs support caching on segment level and request level. https://beta.nextjs.org/docs/data-fetching/caching#graphql-and-cache
But I can't find any cache adaptor configuration to redis. So I assume it's in memory cache. My question is if its in memory cache and I have a high traffic web site. Will the in memory cache lead to OOM of the server?

**Comment by crazyxhz** [comment]
btw. I'm self hosting nextjs

**Comment by crazyxhz** [comment]
help plz

**Comment by nicholasgriffintn** [comment]
Cache memorises the data, it's actually an implementation from React, that they detailed in their original RFC: https://github.com/acdlite/rfcs/blob/first-class-promises/text/0000-first-class-support-for-promises.md#caveat-data-requests-must-be-cached-between-replays

They have yet to expand on how this works though, for this reason cache is super beta.

**Reply by crazyxhz** [reply]
so it's in memory cache right? And in memory cache is not suitable for production ready app

**Reply by nicholasgriffintn** [reply]
I'm not sure how it effects performance yet honestly, I'm not sure anyone has really dug into it as cache is not a documented api yet.

It's actually strange that it's being recommended in a "stable" release actually.

I would guess, that since it is in memory, you probably do want to be careful how much you're using it for, however, there's no reason it shouldn't be used in production as long as you're not over loading it I'd say.
